src/averageAges.js

```diff
@@ -20,6 +20,23 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const filteredMen = people.filter(person => person.sex === 'm'
+    && (century
+      ? Math.ceil(person.died / 100) === century
+      : true));
+
+  const sumOfMenAges = calculateSumOfAges(filteredMen);
+
+  return filteredMen.length ? getAverage(sumOfMenAges, filteredMen.length) : 0;
+}
+
+function calculateSumOfAges(people) {
+  return people.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+}
+
+function getAverage(sum, length) {
```

Comment:
what's the point of the `getAverage` function if we calculate the sum ourselves?